### DISCUSSION TITLE
Sign out of github copilot in Visual Studio 2022

### DISCUSSION THREAD

**Discussion by paulzarlingo** [community/community#14361] [Copilot Conversations] [answered]
I have the same problem as described in this thread:

https://github.com/github/feedback/discussions/6823

Except in Visual studio 2022 instead of VSCode.

Would anyone happen to know how to sign out of github copilot in Visual Studio 2022 so I can sign back in with the correct account? Thanks!

Edit:

I was able to successfully login on another windows account so it must be an extension setting saved somewhere. However, when I uninstall Visual Studio and the copilot plugin and reinstall, the settings persist and it says I don't have access on my primary account.

I plan to keep looking and will update if I find the file / folder / registry key that needs to be deleted.

**Comment by paulzarlingo** [comment, marked as answer]
After some googling I ran across this thread and it fixed the issue. I did not have to reinstall the plugin or Visual Studio for this fix to work. Just restarted Visual Studio.

https://github.com/github/feedback/discussions/14252

navigate to `C:\Users\%USERNAME%\appdata\local\github-copilot` and delete the files `hosts.json` and `terms.json`

**Reply by Shixin-Tang** [reply]
But I use Macbook Pro, and I can not find the files hosts.json and terms.json even after I have signed in to my GitHub account with my GitHub copiolt extension of Azure Data Studio.

**Reply by sambacha** [reply]
```console
cd $HOME/Library/Application\ Support/Code/
```

there you go

**Reply by Shixin-Tang** [reply]
I got the solution from Microsoft Employee. 
the quickest solution here is to delete your profile for Azure Data Studio.

To delete your profile:

Close Azure Data Studio
Find the azuredatastudio folder in the following location:
Windows %APPDATA%\
Mac $HOME/Library/Application Support/
Linux $HOME/.config/
OPTIONAL: before you delete anything, go to the section below to save any settings.
Delete the azuredatastudio folder.
Launch Azure Data Studio. All settings you had previously will be lost. Before you do anything else, try to configure your account for Copilot.
OPTIONAL Step 3 here: If you have any settings, etc. that you want saved, do this first:

Open the Command Palette (CTRL/CMD + SHIFT + P)
Type Profiles, then select Profiles: Export Profile...
In the Profiles window, select the items to export, then the Export button. Save the file on your machine.
Close Azure Data Studio and go to step 4

**Comment by ReviveDigitalTeam** [comment]
THANK YOU

**Comment by shaktals** [comment]
How can this be so absurdly hard? What a shame.

**Comment by jafristr** [comment]
That didn't work for me. What did work was going to Help->Register Visual Studio and the unwanted github account was listed there under All Accounts where I could remove it.

**Reply by JSalazarSoftware** [reply]
This is definitely the easiest way. Thanks for sharing!

**Reply by sssuperMario** [reply]
What a perfect answer！

**Reply by vladcd** [reply]
excellent :)

**Reply by Radwin-CAI** [reply]
Awesome!

**Reply by waqarahmad90111** [reply]
Thanks! You saved a lot of time.

**Comment by bbchuoo** [comment]
2024 my solution 

Select the Account menu in the activity bar, then select Sign Out for the account you are currently signed in with Copilot.
![vscode-accounts-menu-signout](https://github.com/user-attachments/assets/3cb347c9-6676-4a51-99b4-309a7ca31670)

**Comment by akhanalcs** [comment]
Help > About Microsoft Visual Studio > License Status > All Accounts > Remove the GitHub account.

**Comment by AshishYesale7** [comment]
### To Resolve GitHub Copilot Access Issues in VS Code

If you're experiencing issues with **Visual Studio Code** and GitHub Copilot, you can try the following steps to resolve them:

1. **Switch to a different VS Code profile** or **create a new one**.
2. **Log in using your GitHub account** within the selected profile.

Logging in via GitHub is essential, as it links your GitHub Copilot subscription (especially if you have a GitHub Pro account) and enables full access to Copilot features. This workaround often resolves most integration or access issues related to Copilot in VS Code.

---
It helped me : 
You can set Default profile to your **Microsoft account** , 
And other to github ,

<img width="470" alt="Screenshot 2025-06-07 at 12 30 06 AM" src="https://github.com/user-attachments/assets/c678716c-f53e-4d85-98be-c7c7c69d1d0f" />

<img width="414" alt="Screenshot 2025-06-07 at 12 33 05 AM" src="https://github.com/user-attachments/assets/86052bc4-5b8e-43ef-b9ef-70efd77d79ea" />

**Comment by reddy-13** [comment]
<img width="658" height="98" alt="image" src="https://github.com/user-attachments/assets/94159179-f1a7-41a9-b3cc-12251f5ceb9c" />

**Comment by ghost** [comment]
<img width="1162" height="985" alt="image" src="https://github.com/user-attachments/assets/fe962fd4-844b-454f-8eff-64f79224cd0b" />
